ssl/ssl_cert.cc

```diff
@@ -205,8 +213,10 @@ void ssl_cert_clear_certs(CERT *cert) {
 
   cert->x509_method->cert_clear(cert);
 
-  cert->chain.reset();
-  cert->privatekey.reset();
+  for (int i = 0; i < SSL_PKEY_NUM; i++) {
+    cert->cert_privatekeys[i].chain.reset();
+    cert->cert_privatekeys[i].privatekey.reset();
+  }
```

Comment:
Do `x509_leaf` or `x509_chain` need to be cleared, or "down ref'd"?

Reply:
Great question, `cert->x509_method->cert_clear(cert);` above actually does that for us. 
`cert_clear` is a pointer function to `ssl_crypto_x509_cert_clear`, which clears the X509 related structures within it.
https://github.com/aws/aws-lc/blob/ad53db8f1c1753799a13c39d53a9410fe7a49b2b/ssl/ssl_x509.cc#L262-L268